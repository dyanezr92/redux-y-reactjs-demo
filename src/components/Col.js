import React from 'react'
import PropTypes from 'prop-types'

const Col = (props) => (
  <div className={"col-sm-" + props.w} style={ props.style }>
    { props.children }
  </div>
);

Col.propTypes = {
  w: PropTypes.number.isRequired,
  style: PropTypes.object
}

export default Col
