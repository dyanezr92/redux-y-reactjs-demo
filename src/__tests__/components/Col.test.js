import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components

import Col from '../../components/Col';

describe('<Col w="10"/>', () => {
  describe('render()', () => {
    test('renders the col', () => {
      const wrapper = shallow(<Col w={10}/>);
      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe('<Col w="10" style="background: red;"/>', () => {
  describe('render()', () => {
    test('renders the col', () => {
      const wrapper = shallow(<Col w={10} style={{background: "red" }}/>);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
