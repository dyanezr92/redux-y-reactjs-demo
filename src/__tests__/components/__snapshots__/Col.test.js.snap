// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Col w="10" style="background: red;"/> render() renders the col 1`] = `
<div
  className="col-sm-10"
  style={
    Object {
      "background": "red",
    }
  }
/>
`;

exports[`<Col w="10"/> render() renders the col 1`] = `
<div
  className="col-sm-10"
/>
`;
